src/libUtils/DSComposition.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "DSComposition.h"
+
+using namespace std;
+
+void InternalUpdateDSCommitteeComposition(const PubKey& selfKeyPub,
+                                          DequeOfNode& dsComm,
+                                          const DSBlock& dsblock) {
+  LOG_MARKER();
+
+  // Get the map of all pow winners from the DS Block
+  const auto& NewDSMembers = dsblock.GetHeader().GetDSPoWWinners();
+  DequeOfNode::iterator it;
+  unsigned int NumWinners = 0;
+
+  for (const auto& DSPowWinner : NewDSMembers) {
+    // Check if the current pow candidate is an existing DS Committee member.
+    // ('loser') and find its index.
+    for (it = dsComm.begin(); it != dsComm.end(); ++it) {
```

Comment:
use std::find instead of for loop

Reply:
Can't use `std::find` since `dsComm` is a `DequeOfNodes (Pair<PubKey, Peer>)` and `it` may have a zeroed out Peer and `it->first` is a `PubKey`. This causes `no match for ‘operator==’` errors because the type does not match for comparison. `std::find_if` should be used instead but this introduces lambdas which looks even less clear.

Reply:
I made the change to `std::find` initially but reverted it because it breaks the build.

Reply:
I see